scio-core/src/main/scala/com/spotify/scio/util/FunctionsWithSideInput.scala

```diff
@@ -54,4 +54,24 @@ private[scio] object FunctionsWithSideInput {
       c.output(g(c.element(), sideInputContext(c)))
   }
 
+  /**
```

Comment:
Now that we are not exposing this, maybe move this logic to the same place in PairSCollectionFunctions so it's more coherent?